File: Project Academy/12. Pagination-test/java/app/src/main/java/com/dicoding/academies/data/source/local/room/AcademyDatabase.java
package com.dicoding.academies.data.source.local.room;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.dicoding.academies.data.source.local.entity.CourseEntity;
import com.dicoding.academies.data.source.local.entity.ModuleEntity;

@Database(entities = {CourseEntity.class, ModuleEntity.class},
        version = 1,
        exportSchema = false)
public abstract class AcademyDatabase extends RoomDatabase {
    public abstract AcademyDao academyDao();

    private static volatile AcademyDatabase INSTANCE;

    public static AcademyDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            synchronized (AcademyDatabase.class) {
                INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                        AcademyDatabase.class, "Academies.db")
                        .build();

            }
        }
        return INSTANCE;
    }
}

File: Project Academy/09. Room/kotlin/app/src/main/java/com/dicoding/academies/data/source/local/room/AcademyDatabase.kt
package com.dicoding.academies.data.source.local.room

import android.content.Context

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

import com.dicoding.academies.data.source.local.entity.CourseEntity
import com.dicoding.academies.data.source.local.entity.ModuleEntity

@Database(entities = [CourseEntity::class, ModuleEntity::class], version = 1, exportSchema = false)
abstract class AcademyDatabase : RoomDatabase() {
    abstract fun academyDao(): AcademyDao

    companion object {

        @Volatile
        private var INSTANCE: AcademyDatabase? = null

        fun getInstance(context: Context): AcademyDatabase {
            if (INSTANCE == null) {
                synchronized(AcademyDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context.applicationContext,
                                AcademyDatabase::class.java, "Academies.db")
                                .build()
                    }
                }
            }
            return INSTANCE as AcademyDatabase
        }
    }
}